deploy/ci-infra.Jenkinsfile

```diff
@@ -6,10 +6,62 @@ node {
     }
     def legion = load 'deploy/legionPipeline.groovy'
     try {
+        stage('Build') {
+            result = build job: params.BuildLegionJobName, propagate: true, wait: true, parameters: [
+                    [$class: 'GitParameterValue', name: 'GitBranch', value: params.GitBranch],
+                    string(name: 'PyPiRepository', value: params.PyPiRepository),
+                    string(name: 'PyPiDistributionTargetName', value: params.PyPiDistributionTargetName),
+                    string(name: 'DockerRegistry', value: params.DockerRegistry),
+                    string(name: 'JenkinsPluginsRepository', value: params.JenkinsPluginsRepository),
+                    string(name: 'JenkinsPluginsRepositoryStore', value: params.JenkinsPluginsRepositoryStore),
+                    string(name: 'LocalDocumentationStorage', value: params.LocalDocumentationStorage),
+                    booleanParam(name: 'EnableSlackNotifications', value: params.EnableSlackNotifications),
+                    booleanParam(name: 'EnableDockerCache', value: true)
+            ]
+
+            buildNumber = result.getNumber()
+            print 'Finished build id ' + buildNumber.toString()
+
+            // Save logs
+            logFile = result.getRawBuild().getLogFile()
+            sh """
+            cat "${logFile.getPath()}" | perl -pe 's/\\x1b\\[8m.*?\\x1b\\[0m//g;' > build-log.txt 2>&1
+            """
+            archiveArtifacts 'build-log.txt'
+
+            // Copy artifacts
+            copyArtifacts filter: '*', flatten: true, fingerprintArtifacts: true, projectName: 'Build_Legion_Artifacts', selector: specific(buildNumber.toString()), target: ''
```

Comment:
@alexander-semenets Why is the target not defined? What is default value of target?

Reply:
will be reviewed and cleaned up in #649 story